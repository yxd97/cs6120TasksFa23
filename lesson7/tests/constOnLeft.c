#include <stdio.h>
#include <stdlib.h>
// ARGS: 3
int main(int argc, char** argv) {
    if (argc < 2) {
        return 1;
    }
    int x = atoi(argv[1]);
    int r1 = -3 * x ;
    printf("%d x %d = %d\n", -3, x, r1);
    int r2 = 112233 * x;
    printf("%d x %d = %d\n", 112233, x, r2);
    int r3 = 0x00c0ffee * x;
    printf("%d x %d = %d\n", 0x00c0ffee, x, r3);
    int r4 = -1 * x;
    printf("%d x %d = %d\n", -1, x, r4);
    int r5 = -43958 * x;
    printf("%d x %d = %d\n", -43958, x,  r5);
    return 0;
}
