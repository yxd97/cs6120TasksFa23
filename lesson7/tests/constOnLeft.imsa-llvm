; ModuleID = 'tests/constOnLeft.c'
source_filename = "tests/constOnLeft.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@.str = private unnamed_addr constant [14 x i8] c"%d x %d = %d\0A\00", align 1

; Function Attrs: noinline nounwind optnone uwtable
define dso_local i32 @main(i32 noundef %0, ptr noundef %1) #0 {
  %3 = alloca i32, align 4
  %4 = alloca i32, align 4
  %5 = alloca ptr, align 8
  %6 = alloca i32, align 4
  %7 = alloca i32, align 4
  %8 = alloca i32, align 4
  %9 = alloca i32, align 4
  %10 = alloca i32, align 4
  %11 = alloca i32, align 4
  store i32 0, ptr %3, align 4
  store i32 %0, ptr %4, align 4
  store ptr %1, ptr %5, align 8
  %12 = load i32, ptr %4, align 4
  %13 = icmp slt i32 %12, 2
  br i1 %13, label %14, label %15

14:                                               ; preds = %2
  store i32 1, ptr %3, align 4
  br label %267

15:                                               ; preds = %2
  %16 = load ptr, ptr %5, align 8
  %17 = getelementptr inbounds ptr, ptr %16, i64 1
  %18 = load ptr, ptr %17, align 8
  %19 = call i32 @atoi(ptr noundef %18) #3
  store i32 %19, ptr %6, align 4
  %20 = load i32, ptr %6, align 4
  %21 = mul nsw i32 -3, %20
  %22 = shl i32 %20, 0
  %23 = add i32 0, %22
  %24 = shl i32 %20, 2
  %25 = add i32 %23, %24
  %26 = shl i32 %20, 3
  %27 = add i32 %25, %26
  %28 = shl i32 %20, 4
  %29 = add i32 %27, %28
  %30 = shl i32 %20, 5
  %31 = add i32 %29, %30
  %32 = shl i32 %20, 6
  %33 = add i32 %31, %32
  %34 = shl i32 %20, 7
  %35 = add i32 %33, %34
  %36 = shl i32 %20, 8
  %37 = add i32 %35, %36
  %38 = shl i32 %20, 9
  %39 = add i32 %37, %38
  %40 = shl i32 %20, 10
  %41 = add i32 %39, %40
  %42 = shl i32 %20, 11
  %43 = add i32 %41, %42
  %44 = shl i32 %20, 12
  %45 = add i32 %43, %44
  %46 = shl i32 %20, 13
  %47 = add i32 %45, %46
  %48 = shl i32 %20, 14
  %49 = add i32 %47, %48
  %50 = shl i32 %20, 15
  %51 = add i32 %49, %50
  %52 = shl i32 %20, 16
  %53 = add i32 %51, %52
  %54 = shl i32 %20, 17
  %55 = add i32 %53, %54
  %56 = shl i32 %20, 18
  %57 = add i32 %55, %56
  %58 = shl i32 %20, 19
  %59 = add i32 %57, %58
  %60 = shl i32 %20, 20
  %61 = add i32 %59, %60
  %62 = shl i32 %20, 21
  %63 = add i32 %61, %62
  %64 = shl i32 %20, 22
  %65 = add i32 %63, %64
  %66 = shl i32 %20, 23
  %67 = add i32 %65, %66
  %68 = shl i32 %20, 24
  %69 = add i32 %67, %68
  %70 = shl i32 %20, 25
  %71 = add i32 %69, %70
  %72 = shl i32 %20, 26
  %73 = add i32 %71, %72
  %74 = shl i32 %20, 27
  %75 = add i32 %73, %74
  %76 = shl i32 %20, 28
  %77 = add i32 %75, %76
  %78 = shl i32 %20, 29
  %79 = add i32 %77, %78
  %80 = shl i32 %20, 30
  %81 = add i32 %79, %80
  %82 = shl i32 %20, 31
  %83 = add i32 %81, %82
  store i32 %83, ptr %7, align 4
  %84 = load i32, ptr %6, align 4
  %85 = load i32, ptr %7, align 4
  %86 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef -3, i32 noundef %84, i32 noundef %85)
  %87 = load i32, ptr %6, align 4
  %88 = mul nsw i32 112233, %87
  %89 = shl i32 %87, 0
  %90 = add i32 0, %89
  %91 = shl i32 %87, 3
  %92 = add i32 %90, %91
  %93 = shl i32 %87, 5
  %94 = add i32 %92, %93
  %95 = shl i32 %87, 6
  %96 = add i32 %94, %95
  %97 = shl i32 %87, 9
  %98 = add i32 %96, %97
  %99 = shl i32 %87, 10
  %100 = add i32 %98, %99
  %101 = shl i32 %87, 12
  %102 = add i32 %100, %101
  %103 = shl i32 %87, 13
  %104 = add i32 %102, %103
  %105 = shl i32 %87, 15
  %106 = add i32 %104, %105
  %107 = shl i32 %87, 16
  %108 = add i32 %106, %107
  store i32 %108, ptr %8, align 4
  %109 = load i32, ptr %6, align 4
  %110 = load i32, ptr %8, align 4
  %111 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef 112233, i32 noundef %109, i32 noundef %110)
  %112 = load i32, ptr %6, align 4
  %113 = mul nsw i32 12648430, %112
  %114 = shl i32 %112, 1
  %115 = add i32 0, %114
  %116 = shl i32 %112, 2
  %117 = add i32 %115, %116
  %118 = shl i32 %112, 3
  %119 = add i32 %117, %118
  %120 = shl i32 %112, 5
  %121 = add i32 %119, %120
  %122 = shl i32 %112, 6
  %123 = add i32 %121, %122
  %124 = shl i32 %112, 7
  %125 = add i32 %123, %124
  %126 = shl i32 %112, 8
  %127 = add i32 %125, %126
  %128 = shl i32 %112, 9
  %129 = add i32 %127, %128
  %130 = shl i32 %112, 10
  %131 = add i32 %129, %130
  %132 = shl i32 %112, 11
  %133 = add i32 %131, %132
  %134 = shl i32 %112, 12
  %135 = add i32 %133, %134
  %136 = shl i32 %112, 13
  %137 = add i32 %135, %136
  %138 = shl i32 %112, 14
  %139 = add i32 %137, %138
  %140 = shl i32 %112, 15
  %141 = add i32 %139, %140
  %142 = shl i32 %112, 22
  %143 = add i32 %141, %142
  %144 = shl i32 %112, 23
  %145 = add i32 %143, %144
  store i32 %145, ptr %9, align 4
  %146 = load i32, ptr %6, align 4
  %147 = load i32, ptr %9, align 4
  %148 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef 12648430, i32 noundef %146, i32 noundef %147)
  %149 = load i32, ptr %6, align 4
  %150 = mul nsw i32 -1, %149
  %151 = shl i32 %149, 0
  %152 = add i32 0, %151
  %153 = shl i32 %149, 1
  %154 = add i32 %152, %153
  %155 = shl i32 %149, 2
  %156 = add i32 %154, %155
  %157 = shl i32 %149, 3
  %158 = add i32 %156, %157
  %159 = shl i32 %149, 4
  %160 = add i32 %158, %159
  %161 = shl i32 %149, 5
  %162 = add i32 %160, %161
  %163 = shl i32 %149, 6
  %164 = add i32 %162, %163
  %165 = shl i32 %149, 7
  %166 = add i32 %164, %165
  %167 = shl i32 %149, 8
  %168 = add i32 %166, %167
  %169 = shl i32 %149, 9
  %170 = add i32 %168, %169
  %171 = shl i32 %149, 10
  %172 = add i32 %170, %171
  %173 = shl i32 %149, 11
  %174 = add i32 %172, %173
  %175 = shl i32 %149, 12
  %176 = add i32 %174, %175
  %177 = shl i32 %149, 13
  %178 = add i32 %176, %177
  %179 = shl i32 %149, 14
  %180 = add i32 %178, %179
  %181 = shl i32 %149, 15
  %182 = add i32 %180, %181
  %183 = shl i32 %149, 16
  %184 = add i32 %182, %183
  %185 = shl i32 %149, 17
  %186 = add i32 %184, %185
  %187 = shl i32 %149, 18
  %188 = add i32 %186, %187
  %189 = shl i32 %149, 19
  %190 = add i32 %188, %189
  %191 = shl i32 %149, 20
  %192 = add i32 %190, %191
  %193 = shl i32 %149, 21
  %194 = add i32 %192, %193
  %195 = shl i32 %149, 22
  %196 = add i32 %194, %195
  %197 = shl i32 %149, 23
  %198 = add i32 %196, %197
  %199 = shl i32 %149, 24
  %200 = add i32 %198, %199
  %201 = shl i32 %149, 25
  %202 = add i32 %200, %201
  %203 = shl i32 %149, 26
  %204 = add i32 %202, %203
  %205 = shl i32 %149, 27
  %206 = add i32 %204, %205
  %207 = shl i32 %149, 28
  %208 = add i32 %206, %207
  %209 = shl i32 %149, 29
  %210 = add i32 %208, %209
  %211 = shl i32 %149, 30
  %212 = add i32 %210, %211
  %213 = shl i32 %149, 31
  %214 = add i32 %212, %213
  store i32 %214, ptr %10, align 4
  %215 = load i32, ptr %6, align 4
  %216 = load i32, ptr %10, align 4
  %217 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef -1, i32 noundef %215, i32 noundef %216)
  %218 = load i32, ptr %6, align 4
  %219 = mul nsw i32 -43958, %218
  %220 = shl i32 %218, 1
  %221 = add i32 0, %220
  %222 = shl i32 %218, 3
  %223 = add i32 %221, %222
  %224 = shl i32 %218, 6
  %225 = add i32 %223, %224
  %226 = shl i32 %218, 10
  %227 = add i32 %225, %226
  %228 = shl i32 %218, 12
  %229 = add i32 %227, %228
  %230 = shl i32 %218, 14
  %231 = add i32 %229, %230
  %232 = shl i32 %218, 16
  %233 = add i32 %231, %232
  %234 = shl i32 %218, 17
  %235 = add i32 %233, %234
  %236 = shl i32 %218, 18
  %237 = add i32 %235, %236
  %238 = shl i32 %218, 19
  %239 = add i32 %237, %238
  %240 = shl i32 %218, 20
  %241 = add i32 %239, %240
  %242 = shl i32 %218, 21
  %243 = add i32 %241, %242
  %244 = shl i32 %218, 22
  %245 = add i32 %243, %244
  %246 = shl i32 %218, 23
  %247 = add i32 %245, %246
  %248 = shl i32 %218, 24
  %249 = add i32 %247, %248
  %250 = shl i32 %218, 25
  %251 = add i32 %249, %250
  %252 = shl i32 %218, 26
  %253 = add i32 %251, %252
  %254 = shl i32 %218, 27
  %255 = add i32 %253, %254
  %256 = shl i32 %218, 28
  %257 = add i32 %255, %256
  %258 = shl i32 %218, 29
  %259 = add i32 %257, %258
  %260 = shl i32 %218, 30
  %261 = add i32 %259, %260
  %262 = shl i32 %218, 31
  %263 = add i32 %261, %262
  store i32 %263, ptr %11, align 4
  %264 = load i32, ptr %6, align 4
  %265 = load i32, ptr %11, align 4
  %266 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef -43958, i32 noundef %264, i32 noundef %265)
  store i32 0, ptr %3, align 4
  br label %267

267:                                              ; preds = %15, %14
  %268 = load i32, ptr %3, align 4
  ret i32 %268
}

; Function Attrs: nounwind willreturn memory(read)
declare i32 @atoi(ptr noundef) #1

declare i32 @printf(ptr noundef, ...) #2

attributes #0 = { noinline nounwind optnone uwtable "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #1 = { nounwind willreturn memory(read) "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #2 = { "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #3 = { nounwind willreturn memory(read) }

!llvm.module.flags = !{!0, !1, !2, !3, !4}
!llvm.ident = !{!5}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{i32 8, !"PIC Level", i32 2}
!2 = !{i32 7, !"PIE Level", i32 2}
!3 = !{i32 7, !"uwtable", i32 2}
!4 = !{i32 7, !"frame-pointer", i32 2}
!5 = !{!"Ubuntu clang version 17.0.2 (++20230929073225+098e653a5bed-1~exp1~20230929073241.48)"}
