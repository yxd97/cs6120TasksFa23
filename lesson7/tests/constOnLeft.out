-3 x 3 = -9
112233 x 3 = 336699
12648430 x 3 = 37945290
-1 x 3 = -3
-43958 x 3 = -131874
